src/NServiceBus.Core.Tests/Pipeline/Incoming/InvokeHandlerTerminatorTest.cs

```diff
@@ -117,6 +119,26 @@ public class InvokeHandlerTerminatorTest
             Assert.IsTrue(behaviorContext.Extensions.Get<InvokeHandlerTerminator.State>().ScopeWasPresent);
         }
 
+        [Test]
+        public void Handler_should_throw_friendly_invalidoperationexception_if_null_task()
+        {
+            var terminator = new InvokeHandlerTerminator();
+            var messageHandler = CreateMessageHandler((i, m, ctx) => { }, new FakeMessageHandler(), "Handle", () => null);
```

Comment:
`() => null`

Is a bit hard to discover in these tests. How about adding a static method like `HandleWhichReturnsNull` and pass in that as a `Func<Task>?